### PR TITLE
Pac: config, service, sports. + tests

### PR DESCRIPTION
**Pull request by Vargisious**

### CODE REVIEW

File: src/main/java/com/purple/springdemo/service/FootballEquipment.java [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
 import org.springframework.stereotype.Component;
 
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.logging.Logger;
+
 @Component
 @Scope("singleton")
-public class FootballEquipment implements SportService {
+public class FootballEquipment implements SportService, BeanPostProcessor {
```
Comment (Vargisious) [inline src/main/java/com/purple/springdemo/service/FootballEquipment.java]: My bad, forgot to remove the BeanPostProcessor implementation, it'll be removed

File: src/main/java/com/purple/springdemo/BeanPostProcessorImpl.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.purple.springdemo.service.FootballEquipment;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+
+import java.lang.reflect.Field;
+import java.util.logging.Logger;
+
+public class BeanPostProcessorImpl implements BeanPostProcessor {
+
```
Comment (Aleksandr04) [inline src/main/java/com/purple/springdemo/BeanPostProcessorImpl.java]: remove empty line

File: src/main/java/com/purple/springdemo/service/FootballEquipment.java [hunk 3]
```diff
@@ -1,13 +1,35 @@
...
+import org.springframework.beans.factory.config.BeanPostProcessor;
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.logging.Logger;
+
 @Component
 @Scope("singleton")
```
Comment (Aleksandr04) [inline src/main/java/com/purple/springdemo/service/FootballEquipment.java]: remove this line

File: src/main/java/com/purple/springdemo/service/TennisEquipment.java [hunk 4]
```diff
@@ -3,11 +3,32 @@
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.logging.Logger;
+
 @Component
 @Scope("singleton")
```
Comment (Aleksandr04) [inline src/main/java/com/purple/springdemo/service/TennisEquipment.java]: remove

File: src/main/java/com/purple/springdemo/sports/FootballPlayer.java [hunk 5]
```diff
@@ -1,15 +1,26 @@
...
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.logging.Logger;
+
 @Component
 @Scope("prototype")
```
Comment (Aleksandr04) [inline src/main/java/com/purple/springdemo/sports/FootballPlayer.java]: Change it on PROTOTYPE constant

**Pull request merged**